File: apps/commerce-api/src/main/java/com/loopers/infrastructure/inventory/InventoryJpaRepository.java
package com.loopers.infrastructure.inventory;

import java.util.List;
import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Lock;

import com.loopers.domain.inventory.Inventory;

import jakarta.persistence.LockModeType;

public interface InventoryJpaRepository extends JpaRepository<Inventory, Long> {
    @Lock(LockModeType.PESSIMISTIC_WRITE)
    Optional<Inventory> findByProductIdAndProductOptionId(Long productId, Long productOptionId);

    @Lock(LockModeType.PESSIMISTIC_WRITE)
    List<Inventory> findByProductIdInAndProductOptionIdIn(List<Long> productIds, List<Long> productOptionIds);

    List<Inventory> findAllByProductOptionIdIn(List<Long> optionIds);
}

File: apps/commerce-streamer/src/main/kotlin/com/loopers/infrastructure/inventory/InventoryJpaRepository.kt
package com.loopers.infrastructure.inventory

import com.loopers.domain.inventory.Inventory
import org.springframework.data.jpa.repository.JpaRepository

interface InventoryJpaRepository : JpaRepository<Inventory, Long> {
    fun findByProductIdInAndProductOptionIdIn(productIds: List<Long>, productOptionIds: List<Long>): List<Inventory>
}
